pkg/pillar/hypervisor/xen.go

```diff
@@ -829,3 +833,50 @@ func execWithTimeout(command string, args ...string) ([]byte, bool, error) {
 	}
 	return out, true, err
 }
+
+//verifyDomainState: Verifies if a domain is up and running by its state. Domain can have the following states:
+//r - currently running
+//b - blocked, and not running or runnable
+//p - paused
+//s - a shutdown command has been sent, but the domain isn't dying yet
+//c - the domain has crashed
+//d - the domain is dying, but hasn't properly shut down or crashed
+func verifyDomainState(domainName string) error {
+	for i := 2; i > 0; i-- {
+		//getStateCmd: command to filter domain state from 'xl list <domain>' output
+		getStateCmd := fmt.Sprintf("bash -c 'xl list %s' | awk '{print $5}' | awk 'NR==2'", domainName)
```

Comment:
Please don't use bash.  Some day will remove the shells.
Instead parse the xl list output using go code.